hpat/runtests.py

```diff
@@ -1,10 +1,43 @@
+import os
 import unittest
 import hpat.tests
 
+from hpat.tests.tests_config import tests_to_repeat
 
+repeat_test_number = 2
+if 'REPEAT_TEST_NUMBER' in os.environ:
+    repeat_test_number = int(os.environ['REPEAT_TEST_NUMBER'])
+
+""" Decorator to repeat test execution """
+def repeat_test(test):
+    def repeat_test_wrapper(*args, **kwargs):
+        for _ in range(repeat_test_number):
+            test(*args, **kwargs)
+
+    return repeat_test_wrapper
+
+"""
+    Wrap found tests to execute it multiple times using repeat_test decorator
+
+    loadTestsFromModule returns suiteClass with _tests member
+    which contains further suiteClass instanses:
+    hpat_tests = suiteClass(hpat.tests)
+    suiteClass(hpat.tests)._tests = [suiteClass(hpat.tests.TestBasic), suiteClass(hpat.tests.TestDataFrame), ...]
+    suiteClass(hpat.tests)._tests[0] = suiteClass(hpat.tests.TestBasic)
+    suiteClass(hpat.tests.TestBasic)._tests = [TestBasic testMethod=test_array_reduce, ...]
+    test.id() returns the string like hpat.tests.test_basic.TestBasic.test_array_reduce
+"""
 def load_tests(loader, tests, pattern):
     suite = unittest.TestSuite()
-    suite.addTests(loader.loadTestsFromModule(hpat.tests))
+    hpat_tests = loader.loadTestsFromModule(hpat.tests)
+
+    print('Notice: {} tests will be executed {} times'.format(len(tests_to_repeat), repeat_test_number))
+    for i in range(len(hpat_tests._tests)):
+        for j in range(len(hpat_tests._tests[i]._tests)):
+            if hpat_tests._tests[i]._tests[j].id().split('.')[-1] in tests_to_repeat:
+                hpat_tests._tests[i]._tests[j] = repeat_test(hpat_tests._tests[i]._tests[j])
```

Comment:
Do you still think we need a decorator for this? I'not sure but it looks like it could be done simpler (just by a loop).

Reply:
Here the load_tests function returns the suite to be executed by unittest.
So I can add the test to the list multiple times instead of decorator, or now I think that there is more right, but more complicated way - overload loadTestsFromModule unittest method to make it return desired test collection or something like that.
Right now I do not see the other way.
What do you think about it?

Reply:
I don't think that "overload loadTestsFromModule unittest method" is a good way. We need simpler solution. Need to think.